src/plugins/expressions/common/expression_functions/specs/map_column.ts

```diff
@@ -110,7 +110,7 @@ export const mapColumn: ExpressionFunctionDefinition<
       map((rows) => {
         let type: DatatableColumnType = 'null';
         for (const row of rows) {
-          const rowType = getType(row[id]);
+          const rowType = getType(row[id]) as DatatableColumnType;
```

Comment:
should we update getType function to return correct type ?

Reply:
The `string` is a supertype of the `DatatableColumnType `. That's fine to cast it here.